### PR TITLE
open-vm-tools: 11.2.5 -> 11.3.0

### LINKED ISSUE #45751: open-vm-tools doesn't enable display autoscaling on Wayland
## Issue description

When you set `services.vmwareGuest.enable = true` on NixOS and then start a wayland display manager (in my case - sway) display autoscaling should be enabled (i.e. when you resize the VM window on VMWare it should stretch the display resolution out.)

That doesn't happen and when you try to run the `vmware-user-suid-wrapper` or `vmware-user` and instead you get the following error:

```
Xlib:  extension "VMWARE_CTRL" missing on display ":0".
```

### Steps to reproduce

1.) Enable `services.vmwareGuest.enable = true;`
2.) Rebuild `nixos-rebuild switch`
3.) Start sway (or alternative Wayland-Based DM)
4.) Start `vmware-user-suid-wrapper`
5.) See the error above.

## Technical details

```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.17.15, NixOS, 18.09pre149415.8395f9aa85e (Jellyfish)`
 - multi-user?: `yes`
```

### LINKED ISSUE #102388: Mouse stops working when vmware guest is enabled
In a NixOS 20.09 guest on a VMWare Windows host, with dwm as my window manager, the mouse cursor neither moves nor registers clicks when `virtualisation.vmware.guest.enable = true`.

**To Reproduce**
1. Install NixOS 20.09 in VMWare Workstation Player 16 VM (from the minimal ISO)
2. Enable xserver and dwm and `virtualisation.vmware.guest.enable` in `configuration.nix`
3. Run `startx` as normal user with a minimal `~/.xinitrc`

@joamaki, can you have a look?

### PR DESCRIPTION
**Pull request by kjeremy**
###### Motivation for this change

Enables KMS for Wayland and installs the proper udev rules.

Fixes #102388
Fixes #45751

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kjeremy** [comment]
I tested this on nixos 21.05 and it fixed my resolution problems in vmware.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129978` at 42ea61bc run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>open-vm-tools</li>
    <li>open-vm-tools-headless</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/applications/virtualization/open-vm-tools/default.nix:1:62:
```
  |
1 | { stdenv, lib, fetchFromGitHub, makeWrapper, autoreconfHook, fetchpatch,
  |                                                              ^
```
</li>
    <li>warning: unused-argument

Unused argument: `linux-pam`.
Near pkgs/applications/virtualization/open-vm-tools/default.nix:5:3:
```
  |
5 |   linux-pam, libdrm, udev,
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/virtualization/open-vm-tools/default.nix:69:5:
```
   |
69 |     license = licenses.gpl2;
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

rpcsvc-proto is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/virtualization/open-vm-tools/default.nix:24:3:
```
   |
24 |   buildInputs = [ fuse glib icu libdnet libdrm libmspack libtirpc openssl pam procps rpcsvc-proto udev xercesc ]
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/applications/virtualization/open-vm-tools/default.nix [hunk 1]
```diff
@@ -54,24 +38,25 @@ stdenv.mkDerivation rec {
      sed -i 's,/sbin/shutdown,shutdown,' lib/system/systemLinux.c
   '';
 
-  configureFlags = [ "--without-kernel-modules" "--without-xmlsecurity" ]
+  configureFlags = [ "--without-kernel-modules" "--without-xmlsecurity" "--with-udev-rules-dir=${placeholder "out"}/lib/udev/rules.d" ]
     ++ lib.optional (!withX) "--without-x";
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/open-vm-tools/default.nix:42]:
```suggestion
  configureFlags = [
    "--without-kernel-modules"
    "--without-xmlsecurity"
    "--with-udev-rules-dir=${placeholder "out"}/lib/udev/rules.d"
  ] ++ lib.optional (!withX) "--without-x";
```

File: pkgs/applications/virtualization/open-vm-tools/default.nix [hunk 2]
```diff
@@ -1,51 +1,35 @@
...
   nativeBuildInputs = [ autoreconfHook makeWrapper pkg-config ];
-  buildInputs = [ fuse glib icu libdnet libmspack libtirpc openssl pam procps rpcsvc-proto xercesc ]
+  buildInputs = [ fuse glib icu libdnet libdrm libmspack libtirpc openssl pam procps rpcsvc-proto udev xercesc ]
       ++ lib.optionals withX [ gdk-pixbuf-xlib gtk3 gtkmm3 libX11 libXext libXinerama libXi libXrender libXrandr libXtst ];
 
-  patches = [
-    # Fix building with glib 2.68. Remove after next release.
-    # We drop AUTHORS due to conflicts when applying.
-    # https://github.com/vmware/open-vm-tools/pull/505
-    (fetchpatch {
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/open-vm-tools/default.nix:30]: Please remove the unused fetchpatch from inputs.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
@ofborg eval

File: pkgs/applications/virtualization/open-vm-tools/default.nix [hunk 3]
```diff
@@ -1,51 +1,35 @@
-{ stdenv, lib, fetchFromGitHub, makeWrapper, autoreconfHook, fetchpatch,
-  fuse, libmspack, openssl, pam, xercesc, icu, libdnet, procps, libtirpc, rpcsvc-proto,
+{ stdenv, lib, fetchFromGitHub, makeWrapper, autoreconfHook,
+  bash, fuse, libmspack, openssl, pam, xercesc, icu, libdnet, procps, libtirpc, rpcsvc-proto,
   libX11, libXext, libXinerama, libXi, libXrender, libXrandr, libXtst,
   pkg-config, glib, gdk-pixbuf-xlib, gtk3, gtkmm3, iproute2, dbus, systemd, which,
+  libdrm, udev,
   withX ? true }:
```
Comment (SuperSandro2000) [inline pkgs/applications/virtualization/open-vm-tools/default.nix:6]:
```suggestion
{ stdenv, lib, fetchFromGitHub, makeWrapper, autoreconfHook
, bash, fuse, libmspack, openssl, pam, xercesc, icu, libdnet, procps, libtirpc, rpcsvc-proto
, libX11, libXext, libXinerama, libXi, libXrender, libXrandr, libXtst
, pkg-config, glib, gdk-pixbuf-xlib, gtk3, gtkmm3, iproute2, dbus, systemd, which
, libdrm, udev
, withX ? true
}:
```

**Comment by kjeremy** [comment]
Do I have to create a PR to get this into 21.05 or does it roll in eventually?

**Comment by Mic92** [comment]
It does not but only if one adds a lable like this one.

**Comment by ajs124** [comment]
"Update pkgs/applications/virtualization/open-vm-tools/default.nix" I doubt that commit message is according to our contribution guidelines

**Comment by Mic92** [comment]
This seems to originate from github's ui. A squash on merge would have solved that but than also some of the git history would have been lost.

**Comment by Mic92** [comment]
It would be nice if github's ui would allow to amend a commit or create a fixup instead of a new commit.

**Pull request merged**